FprimeZephyrReference/Components/DetumbleManager/BDot.hpp

```diff
@@ -0,0 +1,87 @@
+// ======================================================================
+// \title  BDot.hpp
+// \brief  hpp file for BDot implementation class
+// ======================================================================
+
+#pragma once
+
+#include <array>
+#include <chrono>
+#include <cstdint>
+
+using TimePoint = std::chrono::time_point<std::chrono::steady_clock, std::chrono::microseconds>;
+
+namespace Components {
+
+class BDot {
+  public:
+    // ----------------------------------------------------------------------
+    // Component construction and destruction
+    // ----------------------------------------------------------------------
+
+    //! Construct BDot object
+    BDot();
+
+    //! Destroy BDot object
+    ~BDot();
+
+  public:
+    // ----------------------------------------------------------------------
+    // Public helper methods
+    // ----------------------------------------------------------------------
+
+    // Compute the required dipole moment to detumble the satellite.
+    //
+    // m = -k * (dB/dt) / |B|
+    //
+    // m is the dipole moment in A⋅m²
+    // k is a gain constant
+    // dB/dt is the time derivative of the magnetic field reading in micro-Tesla per second (uT/s)
+    // |B| is the magnitude of the magnetic field vector in micro-Tesla (uT)
```

Comment:
Review with Evan:
Double check units here. Magnetometer reading comments say that it's in gauss

Reply:
Completed this check with @evanjellison